### PR TITLE
configure access log content

### PR DESCRIPTION
**Pull request by BigAirJosh**
nginx access logs are costly, orderweb-web_orderapp_events produces 250k per minute with an annual cost of $30k.  This PR provides an opt in, regex config to configure included access logs requests by status code.

### CODE REVIEW

File: nginx.conf.template [hunk 1]
```diff
@@ -43,10 +43,18 @@ http {
                     'status=$status upstream_status=$upstream_status duration=$request_time size=$body_bytes_sent '
                     '"$remote_user" "$http_referer" "$http_user_agent"';
 
+  # Conditional logging.
+  # To disable 2xx and 3xx logs set LOG_LEVEL=1 TODO how to parameterize?
```
Comment (rfwroo) [inline nginx.conf.template:47]:
```suggestion
  # To disable 2xx and 3xx logs set LOG_LEVEL=1
```

File: start.sh [hunk 2]
```diff
@@ -7,6 +7,7 @@ set -ex
 /bin/sed \
   -e "s/<NGINX_STATUS_PORT>/${NGINX_STATUS_PORT:-81}/g" \
   -e "s:<NGINX_STATUS_ALLOW_FROM>:${NGINX_STATUS_ALLOW_FROM:-all}:g" \
+  -e "s/<NGINX_LOGS_INCLUDE_STATUS_CODE_REGEX>/${NGINX_LOGS_INCLUDE_STATUS_CODE_REGEX:-}/g" \
   -e "s/<NGINX_PORT>/${NGINX_PORT}/g" \
   -e "s/<APP_HOST>/${APP_HOST:-app}/g" \
   -e "s/<APP_PORT>/${APP_PORT}/g" \
```
Comment (rfwroo) [inline start.sh:13]: Hint: this looks like a good use case for [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) (or shameless plug [envfilesubst](https://github.com/rfwatson/envfilesubst))

File: tests/logging.bats [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+@test "with reg ex set to include 200 it logs only 200 requests" {
+  export NGINX_LOGS_INCLUDE_STATUS_CODE_REGEX="^[2]"
+  docker compose -f docker-compose.ci.yml up --detach nginx-test 2> /dev/null
+  docker compose -f docker-compose.ci.yml run wait-test wfi nginx-test:8001 --timeout=60
+
+  curl --head --silent --header "x-set-response-status-code: 200" http://localhost:8001/anything
+
+  run docker compose -f docker-compose.ci.yml logs nginx-test 2> /dev/null
+
+  echo "expected ${lines[-1]} to contain status=200"
```
Comment (rfwroo) [inline tests/logging.bats:36]: Question: I'm loving these tests but also 🤯 - where is `lines` populated here?

**Review by rfwroo — APPROVED** [review]
Awesome work 👏

File: tests/logging.bats (hunk 3 above)
Comment (BigAirJosh) [reply tests/logging.bats:36]: status, lines and output are the bats test output variables from the run command.  we check status for success `[ ${status} -eq 0 ]` output for the full command output and lines when the command outputs more than one line.  [bats docs](https://bats-core.readthedocs.io/en/stable/writing-tests.html#run-test-other-commands)

File: start.sh (hunk 2 above)
Comment (BigAirJosh) [reply start.sh:13]: agreed, didn't pull this in because this was a distraction to the sprint goal ☹️

**Comment by BigAirJosh** [comment]
This is now tested in Orderweb staging.  I have tested both filtering logs to only 4xx and 5xx and leaving all logs enabled regardless of status code.

**Pull request merged**